--- Foundation/library/source/TTSymbolTable.cpp.orig	2014-02-09 18:31:11.000000000 +0100
+++ Foundation/library/source/TTSymbolTable.cpp	2014-02-09 20:42:42.000000000 +0100
@@ -15,6 +15,16 @@
 #include<iostream>
 #include<unordered_map>
 
+//http://stackoverflow.com/questions/21288345/unordered-set-of-pairs-compilation-error
+namespace std {
+    template <>
+    struct hash<TTString> {
+        size_t operator()(const TTString &m) const {
+            // hash method here.
+            return hash<std::string>()(std::string(m.c_str()));
+        }
+    };
+}
 typedef std::unordered_map<TTString, TTSymbolBase*>		TTSymbolTableHash;
 
 
@@ -103,6 +113,8 @@
 #if defined( TT_PLATFORM_WIN ) && OLD
 	return self->lookup(aString.c_str());
 #else
+    if(!mSymbolTable)
+        return 0;
 	TTSymbolTableIter	iter;
 
 	sMutex->lock();
