--- Makefile.common.mac.orig	2014-04-15 06:58:09.000000000 +0200
+++ Makefile.common.mac	2015-10-04 17:25:08.000000000 +0200
@@ -39,8 +39,8 @@
 
 
 # compiler
-CC      ?= clang
-CXX     ?= clang++
+CC       = clang
+CXX      = clang++
 COMPILER = $(CC)
 
 # common compiler flags
@@ -48,5 +48,5 @@
 CXXFLAGS += -DMACOSX -arch $(ARCH) -std=c++0x -stdlib=libc++
 
 # common linker flags
-LDFLAGS  += -L/usr/lib
+LDFLAGS  += -L/opt/local/lib -L/usr/lib
 LDLIBS   += -lpthread -framework IOKit -framework CoreFoundation -framework CoreServices -framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework OpenGL -framework GLUT
