--- headers/graphicstimebox.hpp.orig	2014-02-10 09:50:44.000000000 +0100
+++ headers/graphicstimebox.hpp	2014-02-10 09:51:05.000000000 +0100
@@ -97,7 +97,7 @@
   GraphicsTimeBox(const QPointF &position = QPointF(0,0), const qreal width = 600, const qreal height = 400, QGraphicsItem *parent = 0);
   ~GraphicsTimeBox();
 
-  enum {Type = BoxItemType}; //! Type value for custom item. Enable the use of qgraphicsitem_cast with this item
+  //enum {Type = BoxItemType}; //! Type value for custom item. Enable the use of qgraphicsitem_cast with this item
   virtual int type() const {return Type;}
 
 signals:
