diff --git a/core/vil/CMakeLists.txt b/core/vil/CMakeLists.txt
index 34b44a6..6243b94 100644
--- core/vil/CMakeLists.txt
+++ core/vil/CMakeLists.txt
@@ -61,7 +61,8 @@ ENDIF(ECW_FOUND)
 ##############################
 INCLUDE( ${VXL_CMAKE_DIR}/FindOpenJPEG2.cmake )
 IF(OPENJPEG2_FOUND)
-  INCLUDE_DIRECTORIES(${OPENJPEG2_INCLUDE_DIR})
+  INCLUDE_DIRECTORIES(BEFORE ${OPENJPEG2_INCLUDE_DIR})
+  MESSAGE("OPENJPEG2_INCLUDE_DIR ${OPENJPEG2_INCLUDE_DIR}")
   ADD_DEFINITIONS( ${OPENJPEG2_DEFINITIONS} )
   SET( vil_sources ${vil_sources}
     file_formats/vil_openjpeg.h  file_formats/vil_openjpeg.cxx
@@ -130,7 +131,6 @@ ELSE(VIL_CONFIG_ENABLE_SSE2_ROUNDING)
   SET(VIL_CONFIG_ENABLE_SSE2_ROUNDING 0)
 ENDIF(VIL_CONFIG_ENABLE_SSE2_ROUNDING)
 
-
 # Store the external library configurations in to the config file
 CONFIGURE_FILE( ${vxl_SOURCE_DIR}/core/vil/vil_config.h.in
                 ${vxl_BINARY_DIR}/core/vil/vil_config.h @ONLY IMMEDIATE )
